masonry/src/widget/textbox.rs

```diff
@@ -277,6 +280,21 @@ impl Widget for Textbox {
             width: bc.max().width - 2. * TEXTBOX_MARGIN,
         };
         let size = bc.constrain(label_size);
+        let origin = ctx.widget_state.window_origin();
+        if ctx.widget_state.has_focus {
+            ctx.global_state.signal_queue.push_back(
+                crate::render_root::RenderRootSignal::ImeMoved(
+                    LogicalPosition {
+                        x: origin.x,
+                        y: origin.y + size.height,
+                    },
+                    LogicalSize {
+                        width: size.width,
+                        height: size.height,
+                    },
+                ),
+            );
+        }
```

Comment:
I get the intent, but this code doesn't really work. `layout()` is called *before* a widget is placed. It determines its size, not its position. The way layout currently works, the only point where you know a widget's position for sure is after the entire layout pass has ran.

You can fetch a widget's position from the root using the `find_widget_by_id` methods.